### PR TITLE
feat: implement night mode

### PR DESCRIPTION
**Pull request by smakosh**
## What's new?
Implements night mode and here's how it works:
- Detects and sets your preferable device theme using `prefers-color-scheme` property, the `useMedia` hook lets us access that easier
- But you can still choose which theme you'd like, that stores it on localStorage and will override the first condition

### CODE REVIEW

File: src/@rocketseat/gatsby-theme-docs/components/Logo.js [hunk 1]
```diff
@@ -1,13 +1,32 @@
 import React, { useContext } from "react";
 import styled from "@emotion/styled";
-import Toggle from "../../../components/Toggle";
 import { NightModeContext } from "../../../providers/NightModeProvider";
+import Moon from "../../../components/Icons/Moon";
+import Sun from "../../../components/Icons/Sun";
 import logo from "../assets/izm.png";
 
 const Wrapper = styled.div`
```
Comment (mouradaouinat) [inline src/@rocketseat/gatsby-theme-docs/components/Logo.js:8]: Please make this wrapper 100% width, because it looks a bit off on the preview

**Review by mouradaouinat — CHANGES_REQUESTED** [review]
Great work just one tiny detail 👍

File: src/@rocketseat/gatsby-theme-docs/components/Logo.js (hunk 1 above)
Comment (smakosh) [reply src/@rocketseat/gatsby-theme-docs/components/Logo.js:8]: Done!

**Comment by mouradaouinat** [comment]
LGTM 👍 Good work 🎉

**Pull request merged**